psydac/linalg/stencil.py

```diff
@@ -424,46 +474,47 @@ def __del__(self):
     def space(self):
         return self._space
 
-    #...
-    @property
-    def dtype(self):
-        return self._space.dtype
-
-    #...
-    def dot(self, v):
+    # ...
+    def toarray(self, *, order='C', with_pads=False):
         """
-        Return the inner vector product between self and v.
-
-        If the values are real, it returns the classical scalar product.
-        If the values are complex, it returns the classical sesquilinear product with linearity on the vector v.
+        Return a numpy 1D array corresponding to the given StencilVector,
+        with or without pads.
 
         Parameters
         ----------
-        v : StencilVector
-            Vector of the same space than self needed for the scalar product.
+        with_pads : bool
+            If True, include pads in output array.
+
+        order: {'C','F'}
+             Memory representation of the data ‘C’ for row-major ordering (C-style), ‘F’ column-major ordering (Fortran-style).
 
         Returns
         -------
-        null: self._space.dtype
-            Scalar containing scalar product of v and self.
+        array : numpy.ndarray
+            A copy of the data array collapsed into one dimension.
 
         """
 
-        assert isinstance(v, StencilVector)
-        assert v._space is self._space
+        # In parallel case, call different functions based on 'with_pads' flag
+        if self.space.parallel:
+            if with_pads:
+                return self._toarray_parallel_with_pads(order=order)
+            else:
+                return self._toarray_parallel_no_pads(order=order)
 
-        inner_func = self._space._inner_func
-        inner_args = (self._data, v._data, *self._space._inner_consts)
+        # In serial case, ignore 'with_pads' flag
+        return self.toarray_local(order=order)
```

Comment:
this should be specified in the docstring, no ?

Reply:
Here I have not changed anything from the original code (comments included), just moved things around...

Reply:
Ok, I'll add the comment in the docstring if you agree